ts/src/management/tunnelManagementHttpClient.ts

```diff
@@ -817,8 +817,27 @@ export class TunnelManagementHttpClient implements TunnelManagementClient {
             }
         }
 
+        if (!errorMessage && error.response && error.response.status &&
+            error.response.status >= 400 && error.response.status < 500 &&
+            error.response.headers
+        ) {
+            const headers = error.response.headers;
+            const servedBy = headers['X-Served-By'] || headers['x-served-by'];
+            if (!/tunnel-/.test(servedBy)) {
```

Comment:
I think this might need to be:

/tunnel**s**-/

Reply:
Tested from the browser using these changes in Live Share. Before making the request I changed the authorization header to an invalid entry. This results in a 401 and hits this code path. The `x-served-by` header is as follows:  

![image](https://github.com/microsoft/dev-tunnels/assets/8161247/4bd84697-9104-4830-be93-7b4d9cbf5859)

Reply:
Results in the following errorMessage:
"The tunnel request resulted in 401 status, but the request did not reach the tunnel service. This may indicate the domain 'global.rel.tunnels.api.visualstudio.com' is blocked by a firewall."

Reply:
I can see you have it correct in the C# SDK. Wil make this change in TS.